### DISCUSSION TITLE
poor information how to upgrade ruby to 2.7.2

### DISCUSSION THREAD

**Discussion by jakesullyneytiri** [mastodon/mastodon#27930] [General]
I saw you guys released v3.3.0 but release log is way to poor when it comes to how to upgrade ruby to 2.7.2.

So i would recommend you guys to add short and best way to upgrade it correctly, if it has to be done via root or mastodon user.

Anyway i have found this https://gorails.com/setup/ubuntu/18.04 but confused if i have to do it through mastodon user or root?

So i would like a proper answer to this so it get's upgraded properly and not incorrectly.

**Comment by jakesullyneytiri** [comment]
Update: Found out i only needed to run git -C /home/mastodon/.rbenv/plugins/ruby-build pull and then rvbenv install 2.7.2 to install ruby 2.7.2 and it needed to be done through mastodon user.

this might be good to add to release notes under upgrade part for 3.3.0 so users will know what to do if they are beginners.

**Comment by ariasuni** [comment]
I suppose a link to [“Installing Ruby versions”](https://github.com/rbenv/rbenv#installing-ruby-versions) in the release notes would be appropriate. The documentation [on upgrading](https://docs.joinmastodon.org/admin/upgrading/) already says to switch to user `mastodon` to execute release-specific upgrade instructions, though.

**Comment by bravok** [comment]
This burned me too, during the Mastodon 4.0.1 upgrade off the Digital Ocean droplet.
